### PR TITLE
Implement fmtDuration using Formatter

### PR DESCRIPTION
**Pull request by jdknezek**
Per #8007, this removes single-purpose `std.fmt.Duration` and `std.fmt.duration` and creates `std.fmt.fmtDuration` which uses the `std.fmt.Formatter` pattern.

### CODE REVIEW

File: lib/std/fmt.zig [hunk 1]
```diff
@@ -1261,7 +1261,12 @@ pub fn formatDuration(ns: u64, writer: anytype) !void {
     return;
 }
 
-test "formatDuration" {
+/// Return a Formatter for a number of nanoseconds using `formatDuration`.
+fn fmtDuration(ns: u64) Formatter(formatDuration) {
```
Comment (LemonBoy) [inline lib/std/fmt.zig:1265]:
```suggestion
pub fn fmtDuration(ns: u64) Formatter(formatDuration) {
```

File: lib/std/fmt.zig [hunk 2]
```diff
@@ -1214,7 +1214,7 @@ pub fn formatIntBuf(out_buf: []u8, value: anytype, base: u8, uppercase: bool, op
 ///
 /// - #ns
 /// - [#y][#w][#d][#h][#m]#[.###][u|m]s
-pub fn formatDuration(ns: u64, writer: anytype) !void {
+pub fn formatDuration(ns: u64, comptime fmt: []const u8, options: std.fmt.FormatOptions, writer: anytype) !void {
```
Comment (LemonBoy) [inline lib/std/fmt.zig:1217]: This can be un-pub'd.

File: lib/std/fmt.zig [hunk 3]
```diff
@@ -1288,24 +1293,13 @@ test "formatDuration" {
...
-    pub fn format(self: Duration, comptime fmt: []const u8, options: FormatOptions, writer: anytype) !void {
-        return formatDuration(self.ns, writer);
-    }
-};
-
-/// Formats a number of nanoseconds according to its magnitude. See `formatDuration`.
-pub fn duration(ns: u64) Duration {
-    return Duration{ .ns = ns };
-}
+const Duration = @compileError("Duration has been deprecated; wrap your argument in std.fmt.fmtDuration instead");
```
Comment (LemonBoy) [inline lib/std/fmt.zig:1301]: Since the `Duration` formatter was introduced within this release you can simply delete those.

**Review by LemonBoy — APPROVED** [review]
Great job! I found some minor nits regarding the method visibility.
We could also extract the `Bi` format into its own `Formatter`, that's the last weird case baked into `std.fmt`.

**Comment by jdknezek** [comment]
Thank you for the feedback! Sorry for the sloppy visibility - I only checked the test, not my own code that used it.

**Pull request merged**